tests/api_resources/test_account.py

```diff
@@ -1,148 +1,68 @@
-            None,
-        )
-
-    def test_update_account(self):
-        acct = stripe.Account.construct_from({
-            'id': 'acct_update',
-            'legal_entity': {'first_name': 'Joe'},
-        }, 'api_key')
-        acct.legal_entity['first_name'] = 'Bob'
-        acct.save()
-
-        self.requestor_mock.request.assert_called_with(
-            'post',
-            '/v1/accounts/acct_update',
-            {
-                'legal_entity': {
-                    'first_name': 'Bob',
-                },
-            },
-            None,
-        )
-
-    def test_account_delete_bank_account(self):
-        source = stripe.BankAccount.construct_from({
-            'account': 'acc_delete_ba',
-            'id': 'ba_delete_ba',
-        }, 'api_key')
-        source.delete()
-
-        self.requestor_mock.request.assert_called_with(
-            'delete',
-            '/v1/accounts/acc_delete_ba/external_accounts/ba_delete_ba',
-            {},
-            None
+    def test_is_creatable(self):
+        resource = stripe.Account.create(
+            country='US',
+            type='custom'
         )
-
-    def test_reject_account(self):
-        self.mock_response({
-            'id': 'acct_reject',
-            'verification': {
-                'disabled_reason': 'rejected.fraud'
-            },
-        })
-
-        obj = stripe.Account.construct_from({
-            'id': 'acct_reject'
-        }, 'mykey')
-
-        self.assertTrue(obj is obj.reject(reason='fraud'))
-        self.assertEqual('rejected.fraud', obj.verification['disabled_reason'])
-        self.assertEqual('acct_reject', obj.id)
-
-        self.requestor_mock.request.assert_called_with(
+        self.assert_requested(
             'post',
-            '/v1/accounts/acct_reject/reject',
-            {'reason': 'fraud'},
-            None
+            '/v1/accounts'
         )
+        self.assertIsInstance(resource, stripe.Account)
 
-    def test_reject_account_without_reason(self):
-        self.mock_response({
-            'id': 'acct_reject',
-            'verification': {
-                'disabled_reason': 'rejected.fraud'
-            },
-        })
-
-        obj = stripe.Account.construct_from({
-            'id': 'acct_reject'
-        }, 'mykey')
-
-        self.assertTrue(obj is obj.reject())
-        self.assertEqual('rejected.fraud', obj.verification['disabled_reason'])
-        self.assertEqual('acct_reject', obj.id)
-
-        self.requestor_mock.request.assert_called_with(
+    def test_is_saveable(self):
+        account = stripe.Account.retrieve(TEST_RESOURCE_ID)
+        account.metadata['key'] = 'value'
+        resource = account.save()
+        self.assert_requested(
             'post',
-            '/v1/accounts/acct_reject/reject',
-            {},
-            None
+            '/v1/accounts/%s' % resource.id
         )
+        self.assertIsInstance(resource, stripe.Account)
+        self.assertTrue(resource is account)
```

Comment:
@ob-stripe we should make sure that all calls to save() (on all resources) test this explicitly.

Reply:
Yep. More generally, we should have a similar assert for all instance methods that `return self`.